samza-core/src/test/java/org/apache/samza/task/TestAsyncRunLoop.java

```diff
@@ -39,6 +39,7 @@
 import org.apache.samza.container.TaskInstanceExceptionHandler;
```

Comment:
Can we add the test case with a task with intermediate streams? We should test that the EoS works in the following case:
--topic1--->Task1----output1-->
                   /|\      |
                    |       |
                    Lt3__|

Where t3 is the intermediate topic and EoS is finally triggered.